test/opt/opt_test_common.h

```diff
@@ -0,0 +1,128 @@
+// in all copies or substantial portions of the Materials.
+//
+// MODIFICATIONS TO THIS FILE MAY MEAN IT NO LONGER ACCURATELY REFLECTS
+// KHRONOS STANDARDS. THE UNMODIFIED, NORMATIVE VERSIONS OF KHRONOS
+// SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
+//    https://www.khronos.org/registry/
+//
+// THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+// MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+
+#ifndef LIBSPIRV_TEST_OPT_OPT_TEST_COMMON_H_
+#define LIBSPIRV_TEST_OPT_OPT_TEST_COMMON_H_
+
+#include <memory>
+
+#include <gtest/gtest.h>
+
+#include "source/opt/ir_builder.h"
+#include "source/opt/module.h"
+#include "spirv-tools/libspirv.h"
+
+namespace spvtools {
+namespace opt {
+
+// TODO(antiagainst): expand and export these functions as the C++ interface in
+// libspirv.hpp.
+
+// Assembles the given assembly |text| and returns the binary.
+std::vector<uint32_t> Assemble(const std::string& text) {
+  spv_context context = spvContextCreate(SPV_ENV_UNIVERSAL_1_1);
+  spv_binary binary = nullptr;
+  spv_diagnostic diagnostic = nullptr;
+
+  spv_result_t status =
+      spvTextToBinary(context, text.data(), text.size(), &binary, &diagnostic);
+  EXPECT_EQ(SPV_SUCCESS, status) << "assemble text to binary failed";
+  std::vector<uint32_t> result(binary->code, binary->code + binary->wordCount);
+
+  spvDiagnosticDestroy(diagnostic);
+  spvBinaryDestroy(binary);
+  spvContextDestroy(context);
+
+  return result;
+}
+
+// Skips all leading comment lines and returns the pointer to the start of
+// instructions. This function relies on the fact that a comment line strictly
+// has the prefix of ';' and postfix of '\n'.
+const char* SkipComments(const char* assembly) {
+  while (*assembly == ';') {
+    while (*assembly != '\n') ++assembly;
+    ++assembly;
+  }
+  return assembly;
+}
+
+// Disassembles the given SPIR-V |binary| and returns the assembly.
+std::string Disassemble(const std::vector<uint32_t>& binary) {
+  spv_context context = spvContextCreate(SPV_ENV_UNIVERSAL_1_1);
+  spv_text text = nullptr;
+  spv_diagnostic diagnostic = nullptr;
+
+  spv_result_t status = spvBinaryToText(context, binary.data(), binary.size(),
+                                        0, &text, &diagnostic);
+  EXPECT_EQ(SPV_SUCCESS, status) << "disassemble binary to text failed";
+  std::string result(SkipComments(text->str), text->str + text->length);
+
+  spvDiagnosticDestroy(diagnostic);
+  spvTextDestroy(text);
+  spvContextDestroy(context);
+
+  return result;
+}
+
+spv_result_t SetSpvHeader(void* builder, spv_endianness_t, uint32_t magic,
```

Comment:
I'd prefer to not expose these two functions.  Best to hide them and the body of BuildModule inside a .cpp file.
Also, they need blurbs.